nixos/doc/manual/release-notes/rl-2009.xml

```diff
@@ -400,6 +400,17 @@ systemd.services.nginx.serviceConfig.ReadWritePaths = [ "/var/www" ];
       The rkt module has been removed, it was archived by upstream.
     </para>
    </listitem>
+   <listitem>
+     <para>
+       The FQDN is now added (in addition to the hostname) to
+       <literal>/etc/hosts</literal>. This fixes the FQDN detection (e.g.
+       <literal>hostname --fqdn</literal>) which is required by some
+       applications. These new entries take precedence over entries from the
+       DNS which could cause regressions in some very specific setups.
+       The old behaviour can e.g. be restored by using
+       <literal>networking.hosts = lib.mkForce { "127.0.1.1" = [ config.networking.hostName ]; };</literal>.
```

Comment:
We should also add the IPv6 entry - it's also present in the default installation.

Reply:
But only for `localhost`, not the hostname. The hostname is only resolved to `127.0.1.1` (if I didn't miss anything).

BTW: Should we resolve the FQDN and hostname to `127.0.1.1` (old implementation) or `127.0.0.1` (current state of this PR)? Using `127.0.1.1` has the advantage that we can resolve it back to the FQDN/hostname but also the huge drawback that it won't work if a service only listens to `127.0.0.1` (which is often the case, e.g. for DBs, CUPS, static site generators, etc. - and one can easily miss the fact that the IP isn't exactly `127.0.0.1`).

Due to the drawbacks / potential issues I'm preferring `127.0.0.1` but I don't have a strong opinion. Also if services listen to `::1` in addition to `127.0.0.1` it should still work if we use `127.0.1.1`.

Reply:
We decided to use `127.0.0.2` to match the behaviour of `nss-myhostname` (systemd).